paths/paths.go

```diff
@@ -210,6 +211,15 @@ var drivePrefixes = map[Drive]prefix{
 	TestDrive:         {"./testdata/", "./testdata/", "/dev/null/"},
 }
 
+func SafeParse(ctx workflow.Context, path string) (Path, error) {
+	var p Path
+	err := workflow.SideEffect(ctx, func(ctx workflow.Context) any {
+		return MustParse(path)
```

Comment:
MustParse panics, is that ok?

Reply:
Yes. Basically the idea is that if the workflow is deterministic then the parameters will always be the same so retrying with the same values is useless, thus a panic which is a "non-retryable error" in essence. The panic is caught by temporal so it does not affect the other threads/routines.